frappe/website/doctype/personal_data_download_request/personal_data_download_request.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019, Frappe Technologies and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from frappe.model.document import Document
+
+class PersonalDataDownloadRequest(Document):
+	def after_insert(self):
+		if frappe.session.user in ['Administrator', 'Guest']:
+			frappe.throw(_("This user cannot request to download data"))
+		else:
+			personal_data = get_user_data(frappe.session.user)
+			self.generate_file_and_send_mail(personal_data)
+
+	def generate_file_and_send_mail(self, personal_data):
+		"""generate the file link for download"""
+		user_name = self.user_name.replace(' ','-')
+		f = frappe.get_doc({
```

Comment:
Variables representing Document objects must be slugged versions of the DocType
[example: ``` sales_order = frappe.get_doc('Sales Order', 'SO-0001') ```]

Reply:
f is pretty standard variable for file. Plz check the code for file doctype. Please make sure you check the code and understand before reviewing and asking for changes.

Reply:
i have written this as per the contribution guidelines